web/controllers/v1/token_controller.ex

```diff
@@ -6,23 +6,60 @@ defmodule Cog.V1.TokenController do
   alias Cog.Models.User
   alias Cog.Passwords
 
-  def create(conn, %{"username" => username,
-                     "password" => password}) when not(is_nil(username))
-                                               and not(is_nil (password)) do
+  def create(conn, params) do
+    case validate_params(params) do
+      {:ok, %{"password" => password} = params} ->
+        case find_user(params) do
+          {:ok, user} ->
+            verify_password(user, conn, password)
+          {:error, :not_found} ->
+            Passwords.matches?(nil, "")
+
+            conn
+            |> put_status(:forbidden)
+            |> json(%{errors: "Invalid credentials"})
+        end
+      {:error, :invalid_params} ->
+        conn
+        |> put_status(:unauthorized)
+        |> json(%{errors: "Must supply both username or email and password"})
```

Comment:
It might be less ambiguous to say "both username/email and password"... anytime I see "or"s and "and"s without parentheses, my spidey sense gets triggered 😄